### PR TITLE
Feat/merge slack into main

### PR DESCRIPTION
**Pull request by knopgm**
# Description

This PR implements a Slack bot integration to the backend, enabling the system to send direct messages to users via Slack after payment events. Specifically, users receive a Slack DM for order confirmations or order failures (e.g., payment issues).
The implementation uses the Slack Java SDK (MethodsClient) to open direct message channels and send rich message blocks.
This change also includes comprehensive unit tests for the new SlackService, covering success, error, and edge cases.

# Type of change
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] Unit tests
    - Mocks MethodsClient and injects it via Spring test configuration.
    - Verifies no Slack API calls for null/empty user IDs.
    - Simulates Slack API exceptions and ensures no exception is thrown from the service.
    - Covers all main edge cases for Slack messaging.
- [x] Manual testing
    - Configure a Postman collection with:
        - Register:
Screenshot 2025-08-20 at 23 56 42
-
    -
        - Login:  
Screenshot 2025-08-20 at 23 57 18
-
    -
        -  Create a Product:
Screenshot 2025-08-20 at 23 58 05
-
    -
        -  Create a Cart:
Screenshot 2025-08-20 at 23 58 20
Screenshot 2025-08-20 at 23 58 41
-
    -
        - Place the Order:
Screenshot 2025-08-20 at 23 59 06
Screenshot 2025-08-20 at 23 58 41

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix/feature works
- [x] New and existing tests pass locally with my changes

# Reference
[Slack API](https://api.slack.com/apis/events-api)

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -3,20 +3,26 @@ spring.h2.console.enabled=true
 spring.sql.init.mode=always
 server.port=8080
 server.error.whitelabel.enabled=false
+security.jwt.token.secret-key=
```
[the remark this answers is not in the document]
Comment (knopgm) [reply src/main/resources/application.properties:6]: Keep it empty.

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -3,20 +3,26 @@ spring.h2.console.enabled=true
...
 security.jwt.token.expire-time-millis=3600000
 spring.jpa.hibernate.ddl-auto=create
 spring.jpa.show-sql=true
 spring.jpa.properties.hibernate.format_sql=true
+stripe.secret=
```
[the remark this answers is not in the document]
Comment (knopgm) [reply src/main/resources/application.properties:11]: keep as is.

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -3,20 +3,26 @@ spring.h2.console.enabled=true
...
 spring.mail.port=587
+spring.mail.username=
+spring.mail.password=
 spring.mail.properties.mail.smtp.auth=true
 spring.mail.properties.mail.smtp.starttls.enable=true
 
 spring.jpa.defer-datasource-initialization=true
 spring.sql.init.data-locations=classpath:data.sql
 
+slack.bot.token=
```
[the remark this answers is not in the document]
Comment (knopgm) [reply src/main/resources/application.properties:25]: Keep as is.

**Review by farhanaahmed — APPROVED** [review]
Hello Gabi, 
Nice work on completing the Slack message feature. 🎉 
Thanks for working so hard. 😄 
Please update the PR description photo in the failure section that shows the payment status. 
Everything else seems to be working well, so I'm gonna merge this PR.
